File: web/lib/src/logic/area.dart
import 'package:calculate_engine/src/logic/pi.dart';

// Area of Objects
class Area {
  // Area of a Circle
  double circle(double radius) {
    return pi() * radius * radius;
  }

  // Area of a Cube
  double cube(double length) {
    return length * length * 6;
  }

  // Area of a Cylinder
  double cylinder(double height, double radius) {
    return (pi() * radius * radius + pi() * radius * height) * 2;
  }

  // Area of a Hemisphere
  double hemisphere(double radius) {
    return pi() * radius * radius * 3;
  }

  // Area of a Parallelogram
  double parallelogram(double base, double height) {
    return base * height;
  }

  // Area of a Rectangle
  double rectangle(double length, double width) {
    return length * width;
  }

  // Area of a Sphere
  double sphere(double radius) {
    return pi() * radius * radius * 4;
  }

  // Area of a Square
  double square(double length) {
    return length * length;
  }

  // Area of a Trapezium
  double trapezium(double baseOne, double baseTwo, double height) {
    return (baseOne + baseTwo) * height / 2;
  }

  // Area of a Triangle
  double triangle(double base, double height) {
    return base * height / 2;
  }
}

File: lib/src/logic/math/geometry/area.zig
const pi_value = @import("../constant/pi.zig");

// Area of Objects

// Area of a Circle
pub fn circle_area(radius: f64) f64 {
    return pi_value.pi() * (radius * radius);
}

// Area of a Cube
pub fn cube_area(length: f64, width: f64) f64 {
    return 6 * (length * width);
}

// Area of a Cylinder
pub fn cylinder_area(height: f64, radius: f64) f64 {
    return (2 * height * pi_value.pi() * radius) + (2 * pi_value.pi() * radius * radius);
}

// Area of a Hemisphere
pub fn hemisphere_area(radius: f64) f64 {
    return 3 * (pi_value.pi() * radius * radius);
}

// Area of a Parallelogram
pub fn parallelogram_area(base: f64, height: f64) f64 {
    return base * height;
}

// Area of a Rectangle
pub fn rectangle_area(length: f64, width: f64) f64 {
    return length * width;
}

// Area of a Sphere
pub fn sphere_area(radius: f64) f64 {
    return 4 * (pi_value.pi() * radius * radius);
}

// Area of a Square
pub fn square_area(length: f64, width: f64) f64 {
    return length * width;
}

// Area of a Trapezium
pub fn trapezium_area(baseOne: f64, baseTwo: f64, height: f64) f64 {
    return (1 / 2) * (baseOne + baseTwo) * height;
}

// Area of a Triangle
pub fn triangle_area(base: f64, height: f64) f64 {
    return (1 / 2) * base * height;
}
